packages/financial-templates-lib/src/price-feed/QuandlPriceFeed.js

```diff
@@ -0,0 +1,248 @@
+  }
+
+  async update() {
+    const currentTime = this.getTime();
+
+    // Return early if the last call was too recent.
+    if (this.lastUpdateTime !== undefined && this.lastUpdateTime + this.minTimeBetweenUpdates > currentTime) {
+      this.logger.debug({
+        at: "QuandlPriceFeed",
+        message: "Update skipped because the last one was too recent",
+        currentTime: currentTime,
+        lastUpdateTimestamp: this.lastUpdateTime,
+        timeRemainingUntilUpdate: this.lastUpdateTimes + this.minTimeBetweenUpdates - currentTime
+      });
+      return;
+    }
+
+    this.logger.debug({
+      at: "QuandlPriceFeed",
+      message: "Updating QuandlPriceFeed",
+      currentTime: currentTime,
+      lastUpdateTimestamp: this.lastUpdateTime
+    });
+
+    // Find the closest day that completed before the beginning of the lookback window, and use
+    // it as the start date.
+    const startLookbackWindow = currentTime - this.lookback;
+    const startDateString = this._secondToDateTime(startLookbackWindow);
+    const endDateString = this._secondToDateTime(currentTime);
+
+    // 1. Construct URL.
+    // See https://docs.quandl.com/docs/parameters-2 for how this url is constructed.
+    const url = [
+      `https://www.quandl.com/api/v3/datasets/${this.datasetCode}/${this.databaseCode}/data.json?`,
+      `start_date=${startDateString}&end_date=${endDateString}`,
+      `&collapse=daily&api_key=${this.apiKey}`
+      // Theoretically you could change granularity to be greater than daily but this doesn't seem
+      // useful to implement flexibility for right now.
+    ].join("");
+
+    // 2. Send request.
+    const historyResponse = await this.networker.getJson(url);
+
+    // 3. Check responses.
+    if (
+      !historyResponse?.dataset_data?.data ||
+      historyResponse.dataset_data.data.length === 0 ||
+      historyResponse.dataset_data.data.some(dailyData => dailyData.length === 0)
+    ) {
+      throw new Error(`🚨Could not parse price result from url ${url}: ${JSON.stringify(historyResponse)}`);
+    }
+
+    // 4. Parse results.
+    // Return data structure:
+    // {
+    //   "dataset_data": {
+    //     "data": [
+    //       [
+    //         "2021-03-16",
+    //          1730.2,
+    //          1740.5,
+    //          1724.4,
+    //          1730.1,
+    //          1.7,
+    //          1730.9,
+    //          41843.0,
+    //          21285.0
+    //       ]
+    //      ...more data for different days
+    //     ],
+    //     ...other data we don't care about
+    //   }
+    // }
+    const newHistoricalPricePeriods = historyResponse.dataset_data.data
+      .map(dailyData => ({
+        // Output data should be a list of objects with only the open and close times and prices.
+        // Note: Data is formatted as [Date, Open, High, Low, Last, Change, Settle, Volume, Previous Day Open Interest]
+        openTime: this._dateTimeToSecond(dailyData[0]) - 24 * 3600,
```

Comment:
i dont think this is right, since you are converting this with moment, it would be giving you the start of the day.  you would have to add this to close time, you probably want something like 

`moment(time,'YYYY-MM-DD').endOf('day').unix()`

Reply:
Aha thats a really good catch. Also need to fix this in the ForexDailyPriceFeed as well.